lib/block_view.go

```diff
@@ -3593,6 +3625,97 @@ func (bav *UtxoView) _connectTransaction(txn *MsgDeSoTxn, txHash *BlockHash,
 	return utxoOpsForTxn, totalInput, totalOutput, fees, nil
 }
 
+// _connectFailingTransaction is used to process the fee and burn associated with the user submitting a failing transaction.
+// A failing transaction passes formatting validation, yet fails connecting to the UtxoView. This can happen for a number
+// of reasons, such as insufficient DESO balance, wrong public key, etc. With Revolution's Fee-Time block ordering, these
+// failing transactions are included in the blocks and their fees are burned. Initially blocks will include full transactions,
+// and in the future iterations blocks will only include the failing transaction hashes. This is done to prevent spamming
+// attacks on the network with seemingly valid transactions that never commit and just end up occupying the mempool space.
+// In addition, a major part of the effective fees of this transaction is burned with BMF. This makes spam attacks economically
+// disadvantageous. Attacker's funds are burned, to the benefit of everyone else on the network. BMF algorithm also computes
+// a utility fee, which is distributed to the block producer.
+func (bav *UtxoView) _connectFailingTransaction(txn *MsgDeSoTxn, blockHeight uint32, verifySignatures bool) (
```

Comment:
What if the transaction correctly connects? Where do you intend to verify that it's not a connecting txn?

Reply:
This would be a part of the block producer. A transaction will be first connected, as usual. Upon failure, the transaction will be connected with the `_connectFailingTransaction` function.